### PR TITLE
Gracefully handle proc_open failing.

### PR DESCRIPTION
**Pull request by joehoyle**
Fixes https://github.com/humanmade/Cavalcade-Runner/issues/45.

### CODE REVIEW

File: inc/class-runner.php [hunk 1]
```diff
@@ -128,6 +128,7 @@ public function run() {
 			try {
 				$this->run_job( $job );
 			} catch ( Exception $e ) {
+				trigger_error( sptrinf( 'Unable to run job due to exception: %s', $e->getMessage() ), E_USER_WARNING );
```
Comment (rmccue) [inline inc/class-runner.php]: `sptrinf`, nice work. Guessing you didn't test this then? 😉

File: inc/class-runner.php [hunk 2]
```diff
@@ -274,6 +274,8 @@ protected function run_job( $job ) {
 		$process = proc_open( $command, $spec, $pipes, $cwd );
 
 		if ( ! is_resource( $process ) ) {
+			// Set the job to failed as we don't know if the process was able to run the job.
+			$job->mark_failed( 'Unable to prod_open.' );
 			throw new Exception();
```
Comment (rmccue) [inline inc/class-runner.php]: We could set the exception message here and use that in the `trigger_error` instead, I was just lazy writing this.

File: inc/class-runner.php (hunk 1 above)
Comment (joehoyle) [reply inc/class-runner.php]: May have added the sprintf later :P

**Comment by joehoyle** [comment]
@rmccue you ok to pick this one up? kthnxbye

**Pull request merged**